stack/building-with-pip.rst

```diff
@@ -0,0 +1,241 @@
+For example, in ``daf_butler`` there is this configuration:
+
+.. code-block:: toml
+
+    [tool.setuptools.packages.find]
+    where = ["python"]
+
+    [tool.setuptools]
+    zip-safe = true
+    license-files = ["COPYRIGHT", "LICENSE", "bsd_license.txt", "gpl-v3.0.txt"]
+
+    [tool.setuptools.package-data]
+    "lsst.daf.butler" = ["py.typed", "configs/*.yaml", "configs/*/*.yaml"]
+
+    [tool.setuptools.dynamic]
+    version = { attr = "lsst_versions.get_lsst_version" }
+
+This tells ``setuptools`` that the python files are in a ``python/`` directory and what additional non-python files should be included in the distribution.
+
+The license-files section should reflect the specific needs of your package.
+
+When making a `PyPI`_ distribution, the package should work without relying on the EUPS ``$PACKAGE_DIR`` variable being set.
+This means that any supplementary data such as those that would go in a ``config/`` or ``policy/`` directory should instead be included inside the ``python/`` directory and be accessed using the standard package resources APIs.
+These files must then be listed explicitly in the ``package-data`` section of the configuration file.
+
+.. warning::
+
+  Currently ``pex_config`` does not understand how to read a config from a package using package resources.
+  If configs are to be read they can not be read using the usual ``lsst.utils.getPackageDir`` API and must instead use `importlib.resources` APIs directly.
+  We are planning to make this simpler by adding native support into ``pex_config``.
+
+Using GitHub Actions
+====================
+
+If a package is pip-installable it is likely that you will want to build the package in a GitHub action and run the associated tests.
+If your package depends on other science pipelines packages you will want to install those directly from GitHub from the ``main`` branch since there is no guarantee that `PyPI`_ will have the right version.
+The easiest way to do this is to write a ``requirements.txt`` file which has the direct dependencies that should be installed by the build script.
+This file is a simple text file listing packages and versions.
+
+For example, the ``requirements.txt`` in the ``daf_relation`` package looks like:
+
+.. code-block::
+
+  git+https://github.com/lsst/utils@main#egg=lsst-utils
+  sqlalchemy >= 1.4
+
+The first line tells ``pip`` to install the dependency directly from GitHub.
+The second line is a standard `PyPI`_ dependency.
+These can be installed by running:
+
+.. code-block:: bash
+
+  $ pip install -r requirements.txt
+
+and then the package can be installed with:
+
+.. code-block:: bash
+
+  $ pip install --no-deps .
+
+Where this will skip the dependency check and install the package directly.
+When developing multiple packages at the same time it is possible to change the ``requirements.txt`` file to point at a specific ticket branch rather than ``main``.
+There are checkers available that can block merging if such a change has been made.
```

Comment:
Where are these checkers?

Reply:
At the moment not reusable workflows. They are in the middleware packages.